src/features/step_definitions/legal-agreement.steps.js

```diff
@@ -0,0 +1,91 @@
+const { When, Then } = require("@wdio/cucumber-framework");
+const uploadPage = require("../page_objects/legal_agreement/legal-agreement-upload.page");
+const CheckPage = require("../page_objects/legal_agreement/legal-agreement-check.page");
+
+const path = require("path");
+
+let filename = "";
+let filePath = "";
+let remoteFilePath = "";
+
+When("I choose and upload a {string}", async (document) => {
+  switch (document) {
+    case "legal agreement": {
```

Comment:
You only have one case and no default, do you have any linting rules? I'm sure this would be frowned on, I appreciate you'll look to fill this out when you have more files to upload.

Reply:
removed the case statement until other instances need to be tested